core-build/gulp-core-build/src/logging.ts

```diff
@@ -272,20 +271,33 @@ function _writeTaskError(e: any): void {
 }
 
 function exitProcess(errorCode: number): void {
-
   if (!localCache.watchMode) {
     process.stdout.write('', () => {
       process.exit(errorCode);
     });
   }
 }
 
-function wireUpProcessErrorHandling(): void {
+function wireUpProcessErrorHandling(shouldWarningsFailBuild: boolean | undefined): void {
   if (!wiredUpErrorHandling) {
     wiredUpErrorHandling = true;
+
+    let wroteToStdErr: boolean = false;
+
+    if (shouldWarningsFailBuild) {
+      const oldStdErr: Function = process.stderr.write;
+      // tslint:disable-next-line:no-function-expression
+      process.stderr.write = function (text: string | Buffer): boolean {
+        if (!!text.toString()) {
+          wroteToStdErr = true;
+          return oldStdErr.apply(process.stderr, arguments);
```

Comment:
>arguments [](start = 49, length = 9)

should I store the text so we can print it out in the summary?

Reply:
Note that Rush will already read this, but gulp currently does not print it out in the summary

---
In reply to: [162230487](https://github.com/Microsoft/web-build-tools/pull/495#discussion_r162230487) [](ancestors = 162230487)

Reply:
we decided this was out of scope

---
In reply to: [162230610](https://github.com/Microsoft/web-build-tools/pull/495#discussion_r162230610) [](ancestors = 162230610,162230487)